src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -4156,6 +4158,16 @@ void EmitSingleCharRepeater(RegexNode node, bool emitLengthChecksIfRequired = tr
 
                     sliceStaticPos += iterations;
                 }
+
+                static bool CanUseIndexOfAnyValues(RegexNode node)
+                {
+                    const int MinLengthForVectorizedIndexOfAnyValues = 8;
+
+                    return node.N >= MinLengthForVectorizedIndexOfAnyValues
+                        && node.IsSetFamily
+                        && RegexCharClass.TryGetAsciiSetChars(node.Str!, out char[]? asciiChars)
+                        && asciiChars.Length > 3; // We use 'IndexOfAny(char, char, char)' overloads for shorter sets.
```

Comment:
Is this going to possible result in us allocating more char[]s because for any such set we're going to be allocating a char[] for sets of length 2 or 3 and for no benefit?

Reply:
Yes. Do you think it's worth adding a separate helper (e.g. `GetAsciiSetCharsLength`) to avoid it?